torch/_subclasses/meta_utils.py

```diff
@@ -406,9 +406,12 @@ def is_c_of_r(complex_dtype, real_dtype):
 
                 else:
                     is_leaf = safe_is_leaf(t)
-                    sizes, strides, storage_offset = sym_sizes_strides_storage_offset(
-                        t, source
-                    )
+                    if not t.is_nested:
+                        (
+                            sizes,
+                            strides,
+                            storage_offset,
+                        ) = sym_sizes_strides_storage_offset(t, source)
```

Comment:
Hmmm, what do you think of rearranging so that `def sym_sizes_strides_storage_offset(x, x_src)` checks if `x` is nested, and does the special handling in there (and maybe calls `NestedTensor.create_raggedness_id()` or something).

One reason this is annoying to reason about is because you are now not creating `sizes/strides` at all at the top-level, but there's a bunch of code (line 492 and below) that expects `sizes/strides` to exist, so it can detect happy-path code (like when we're fakeifying a tensor with 0 storage_offset).

Another thing is that doing the NestedTensor logic inside of that function should more naturally support composing subclasses (say we have `TwoTensor(NestedTensor, NestedTensor)` - it would be nice if inner tensors that are nested get fakefied the right way automatically).

Reply:
Good point, I didn't think about what would happen when subclasses compose. Updating the logic to happen inside `sym_sizes_strides_storage_offset`.